### PR TITLE
Add graph of prerequisites in section info card

### PR DESCRIPTION
**Pull request by io12**

### CODE REVIEW

**Comment by io12** [comment]
It looks like the build is failing because prereq_graph.json doesn't exist yet in quacs-data. Should I split this PR into one for the scraper and another for the site changes?

**Comment by elihschiff** [comment]
I would make a pr with the missing file to the data repo. I will merge it. Then you can rerun the tests for this pr.

**Comment by io12** [comment]
https://github.com/quacs/quacs-data/pull/5

File: site/src/components/PrereqGraph.vue [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    border: string;
+  };
+}
+
+@Component({
+  components: {
+    Network,
+  },
+  computed: {
+    options: function () {
```
Comment (benjaminrsherman) [inline site/src/components/PrereqGraph.vue:51]: Is this function able to be moved into the class?

File: site/src/components/sections/SectionInfo.vue [hunk 2]
```diff
@@ -57,6 +62,10 @@ export default class SectionInfo extends Vue {
   get modalTitle(): string {
     return `Section Info: ${this.section.sec} - ${this.section.title}`;
   }
+
+  get course(): string {
```
Comment (benjaminrsherman) [inline site/src/components/sections/SectionInfo.vue:66]: Could you rename this function to `courseCode` to be a bit more descriptive?

**Review by benjaminrsherman — CHANGES_REQUESTED** [review]

**Comment by benjaminrsherman** [comment]
Extra comments after testing the site myself: Can you add a header above the graph named something like "Visualize Prerequisites:"?  I think that'll make it easier to understand what is being shown.  Similarly, I think it'd be better to hide the graph altogether if there are no prerequisites, otherwise it might be a little confusing.

Unrelated to the above, but is it feasible to include reverse-dependencies as well (with the current course still highlighted)?  I'm not sure if the chart would become unreadable for courses with a ton of reverse dependencies (e.g. CSCI-2300).

File: site/src/components/sections/SectionInfo.vue (hunk 2 above)
Comment (io12) [reply site/src/components/sections/SectionInfo.vue]: Fixed

File: site/src/components/PrereqGraph.vue (hunk 1 above)
Comment (io12) [reply site/src/components/PrereqGraph.vue]: Yes, fixed

**Comment by io12** [comment]
Ok, there is now a header and it auto-hides if there are no prereqs.

> Unrelated to the above, but is it feasible to include reverse-dependencies as well (with the current course still highlighted)? I'm not sure if the chart would become unreadable for courses with a ton of reverse dependencies (e.g. CSCI-2300).

The prereq_graph.json is currently an adjacency list, so this would require updating the scraper to output a copy of the graph with the edges flipped. It probably would also be unreadable and have bad performance with lots of courses. A prereq_graph.json with inverted edges might be useful for showing reverse prereqs in a non-graph text format, though.

**Comment by io12** [comment]
Any updates on this? Is there anything blocking merging?

**Comment by benjaminrsherman** [comment]
Oops, I didn't realize that you implemented the changes requested on discord.

**Review by benjaminrsherman — APPROVED** [review]

**Pull request merged**